### ISSUE TITLE
reportUnnecessaryComparison fails to detect `case _:` as unnecessary

### ISSUE THREAD

**Issue by Ten0** [microsoft/pyright#9839]
**Describe the bug**
As of Pyright 1.1.391 (or Pylance 2025.2.1, which is the trigger to me noticing this), `reportUnnecessaryComparison` fails to detect `case _:` as unnecessary.

This makes it impossible to use typing to detect non-exhaustive matches, while also not leaving silent holes in the match if type annotations were not completely accurate.

**Code or Screenshots**

```jsonc
// pyrightconfig.json
{
    "reportUnnecessaryTypeIgnoreComment": "warning",
    "reportUnnecessaryComparison": "warning",
    "reportMatchNotExhaustive": "error",
}
```

Python:
```python
from typing import Literal, TypeAlias

State: TypeAlias = Literal["A", "B", "C"]


def state() -> State:
    return "A"


def f():
    s = state()
    match s:
        case "A" | "B":
            return True
        case "C":
            return False
        case _:  # pyright: ignore[reportUnnecessaryComparison]
            raise ValueError(f"Unexpected state: {s}")
```

This used to not give a warning, but would give one when removing `| "B"` from the first `case`: `Unnecessary "# pyright: ignore" rule: "reportUnnecessaryComparison"`, as indeed the typer would notice that  `case _` would now get used.

This pattern was very useful because it allowed detecting matches that typing thought were non-exhaustive, while also not leaving silent holes in the `match` if type annotations were missing cases (one would obtain silent holes if not adding a `case _:` to the match, because in that case Python just jumps to instructions following the match statement).

Because `case _` seems to now be ignored for `reportUnnecessaryComparison`, it's not possible to detect non-exhaustive matches while still having a `case _` safeguard:
`Unnecessary "# pyright: ignore" rule: "reportUnnecessaryComparison"` always triggers (or if you remove it you never get a warning, regardless of whether there are cases that the typer knows will be handled by `case _` or not).

**VS Code extension or command-line**
Pylance 2025.2.1 / Pyright 1.1.391

**Comment by Ten0** [comment]
Hmm, looks like basedpyright has an implementation for this that would let both work:
https://github.com/DetachHead/basedpyright/pull/593/files
